litmus-portal/frontend/src/pages/MonitoringDashboardPage/index.tsx

```diff
@@ -102,6 +290,108 @@ const DashboardPage: React.FC = () => {
+  useQuery<PrometheusResponse, PrometheusQueryVars>(PROM_QUERY, {
+    variables: {
+      prometheusInput: prometheusQueryData?.promInput ?? {
+        url: '',
+        start: '',
+        end: '',
+        queries: [],
+      },
+    },
+    fetchPolicy: 'no-cache',
+    skip: prometheusQueryData?.promInput.url === '',
+    onCompleted: (eventData) => {
+      let chaosDataUpdates: ChaosDataUpdates = {
+        queryIDs: [],
+        chaosData: [],
+        reGenerate: false,
+        latestEventResult: [],
+      };
+      if (eventData && analyticsData) {
+        const selectedEndTime: number =
+          new Date(moment(selectedDashboard.range.endDate).format()).getTime() /
+          1000;
+        const selectedStartTime: number =
+          new Date(
+            moment(selectedDashboard.range.startDate).format()
+          ).getTime() / 1000;
+        chaosDataUpdates = chaosEventDataParserForPrometheus(
+          prometheusQueryData?.numOfWorkflows,
+          analyticsData,
+          eventData,
+          prometheusQueryData?.chaosEvents,
+          selectedStartTime,
+          selectedEndTime
+        );
+        if (
+          chaosDataUpdates.reGenerate &&
+          !prometheusQueryData.firstLoad &&
+          !selectedDashboard.forceUpdate
+        ) {
+          clearTimeOuts().then(() => {
+            dashboard.selectDashboard({
+              forceUpdate: true,
+            });
+          });
+        }
+        if (!chaosDataUpdates.reGenerate) {
+          if (selectedDashboard.forceUpdate) {
+            dashboard.selectDashboard({
+              forceUpdate: false,
+            });
+          }
+          if (!selectedDashboardInformation.selectionOverride) {
+            setChaosDataSet({
+              ...chaosDataSet,
+              queryIDs: chaosDataUpdates.queryIDs,
+              chaosData: chaosDataUpdates.chaosData,
+              visibleChaos: chaosDataUpdates.chaosData,
+              latestEventResult: chaosDataUpdates.latestEventResult,
+            });
+          } else {
+            setChaosDataSet({
+              ...chaosDataSet,
+              queryIDs: chaosDataUpdates.queryIDs,
+              chaosData: chaosDataUpdates.chaosData,
+              latestEventResult: chaosDataUpdates.latestEventResult,
+            });
+          }
+        }
+        chaosDataUpdates = {
+          queryIDs: [],
+          chaosData: [],
+          reGenerate: false,
+          latestEventResult: [],
+        };
+      }
+    },
+    onError: (error: ApolloError) => {
+      if (
+        error.message ===
+        `bad_data: exceeded maximum resolution of 11,000 points per timeseries. Try decreasing the query resolution (?step=XX)`
+      ) {
+        if (selectedDashboard.refreshRate !== 2147483647) {
+          dashboard.selectDashboard({
+            refreshRate: 2147483647,
+          });
+        }
+        setPrometheusQueryData({ ...prometheusQueryData, firstLoad: true });
+      }
```

Comment:
please keep in the error.message on line 383 in some constant. The same should be done for 214... constant